src/Jupyter/Extensions.cs

```diff
@@ -23,6 +23,21 @@ public static void AddIQSharpKernel(this IServiceCollection services)
         {
             services.AddSingleton<ISymbolResolver, Jupyter.SymbolResolver>();
             services.AddSingleton<IExecutionEngine, Jupyter.IQSharpEngine>();
+            services.AddSingleton<IConfigurationSource, Jupyter.IQSharpEngine>(
+                provider => provider.GetService<IExecutionEngine>() as IQSharpEngine
```

Comment:
I'm not sure we should have the `IQSharpEngine` double as both, the `IExecutionEngine` engine and the `IConfigurationSource`.
I recommend just create a simple configuration class.

Reply:
Good point, I'll split that out. Thankfully shouldn't require any changes to the consumers thanks to dependency injection.